frodo.js

```diff
@@ -1,32 +1,30 @@
 /**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.